plugin/sycl/common/hist_util.cc

```diff
@@ -271,11 +367,48 @@ ::sycl::event BuildHistDispatchKernel(
 
   // force_atomic_use flag is used only for testing
   use_atomic = use_atomic || force_atomic_use;
+
   if (!use_atomic) {
+    const size_t th_block_size = 256;
+    const auto block_params = GetBlocksParameters(qu, size, hist_buffer->Size() / (nbins * 2));
+    const size_t block_size = block_params.block_size;
+    const size_t max_num_bins = gmat.max_num_bins;
+    using GradientPairT = xgboost::detail::GradientPairInternal<FPType>;
+
+    int eu_l1_size = 0;
+    int eu_registers_size = 0;
+    if (true) {
+      // Xe-HPC
+      eu_l1_size = (512 / 8) * 1024;
+      eu_registers_size = (256 / 8) * 1024;
+    } else if (false) {
+      // Xe2-HPG; BMG
+      eu_l1_size = (256 / 8) * 1024;
+      // L1 and registers share the same block
+      eu_registers_size = 0;
+    } else if (false) {
+      // Xe2-HPG; LNL
+      eu_l1_size = (196 / 8) * 1024;
+      // L1 and registers share the same block
+      eu_registers_size = 0;
+    }
+    const int eu_sram_size = eu_l1_size + eu_registers_size;
+    const int buff_size = nbins * sizeof(GradientPairT);
+
+    bool use_local_hist = (buff_size < 0.8 * eu_sram_size)
+                       && (max_num_bins == 256)
```

Comment:
This `256` here is related to `constexpr int kMaxNumBins = 256;` in the kernel. Is it possible to define this constant once and use in both cases?
Because the way it is written now might lead to the issues due to unsynchronized changes of those two pieces of code in the future.